src/main/java/frc/robot/subsystems/drivetrain/SwerveModule.java

```diff
@@ -130,20 +169,36 @@ public SwerveModulePosition getState() {
      * @param state The desired state of the module as a `SwerveModuleState`.
      */
     public void setDesiredState(SwerveModuleState state) {
-        SwerveModuleState optimized = SwerveModuleState.optimize(state, getAngle());
+        Rotation2d currentAngle = getAngle();
+        SwerveModuleState optimized = SwerveModuleState.optimize(state, currentAngle);
+
+        double currentVelocity = driveEncoder.getVelocity();
+        double targetAngle = optimized.angle.getRadians() - offsetRads;
+
+        // Set shuffleboard debug info
+        targetVelEntry.setDouble(optimized.speedMetersPerSecond);
+        currentVelEntry.setDouble(currentVelocity);
+        velErrorEntry.setDouble(optimized.speedMetersPerSecond - currentVelocity);
+
+        targetAngleEntry.setDouble(Math.toDegrees(MathUtil.angleModulus(targetAngle)));
+        currentAngleEntry.setDouble(currentAngle.minus(new Rotation2d(offsetRads)).getDegrees());
+        angleErrorEntry.setDouble(MathUtil.angleModulus(targetAngle - currentAngle.getRadians() + offsetRads));
+
         // driveMotor.set(ControlMode.Velocity, optimized.getFirst() / (DRIVE_TICKS_TO_METERS * 10.0));
-        driveMotor.set(optimized.speedMetersPerSecond); // Only while the module is in percent output
-        steerPidController.setReference(optimized.angle.getRadians() - offsetRads, ControlType.kPosition);
+        drivePidController.setReference(optimized.speedMetersPerSecond, ControlType.kVelocity);
+        // driveMotor.set(optimized.speedMetersPerSecond);
```

Comment:
i think either remove or have some boolean flag to switch btwn velo- and power-based control